### PR TITLE
chore: recorder is trace viewer experiment

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: packages/trace-viewer/src/ui/sourceTab.tsx [hunk 1]
```diff
@@ -66,7 +66,9 @@ export const SourceTab: React.FunctionComponent<{
     highlight.push({ line: targetLine, type: 'running' });
 
     // After the source update, but before the test run, don't trust the cache.
-    if (source.content === undefined || shouldUseFallback) {
+    if (fallbackLocation?.source?.content !== undefined) {
+      source.content = fallbackLocation.source.content;
```
Comment (dgozman) [inline packages/trace-viewer/src/ui/sourceTab.tsx:70]: I don't quite understand this one. Do we show the fallback source even when we have the real source content?

File: packages/playwright-core/src/server/recorder/recorderInTraceViewer.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+async function openApp(trace: string, options?: TraceViewerServerOptions & { headless?: boolean }) {
+  const server = await startTraceViewerServer(options);
+  await installRootRedirect(server, [trace], { ...options, webApp: 'recorder.html' });
+  const page = await openTraceViewerApp(server.urlPrefix('precise'), 'chromium', options);
+  page.on('close', () => gracefullyProcessExitDoNotHang(0));
+}
+
+class RecorderTransport implements Transport {
```
Comment (dgozman) [inline packages/playwright-core/src/server/recorder/recorderInTraceViewer.ts:82]: Why do we need an empty transport?

**Review by dgozman — APPROVED** [review]

File: packages/trace-viewer/src/ui/sourceTab.tsx (hunk 1 above)
Comment (pavelfeldman) [reply packages/trace-viewer/src/ui/sourceTab.tsx:70]: Yes, fallbackLocation is a priority.

File: packages/playwright-core/src/server/recorder/recorderInTraceViewer.ts (hunk 2 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/server/recorder/recorderInTraceViewer.ts:82]: That's our way of communicating to the front-end (trace viewer). Commands from recorder front-end will arrive over this transport.

**Pull request merged**